ext/dom/tests/bug80268.phpt

```diff
@@ -0,0 +1,13 @@
+--TEST--
+Bug #80268 (loadHTML() truncates at NUL bytes)
+--SKIPIF--
+<?php require_once('skipif.inc'); ?>
+--FILE--
+<?php
+$doc = new DOMDocument;
+$doc->loadHTML("<p>foobar</p>");
```

Comment:
If this is supposed to include a null byte, can you please write it as `\0`?

Reply:
Nope, that was a bad test case. And actually my analysis in the commit message was wrong as well:

Actually, libxml does not replace NUL bytes with spaces, but rather
truncates text content at NUL bytes, but generally continues parsing.
Anyhow, `loadHTML()` and `loadHTMLFile()` should behave the same in
this regard.

Reply:
I agree that these should behave consistently, but it would be good to also cross check the behavior of other libxml extensions like XmlReader. I believe at least one of them currently throws a "must not contain null bytes" error in this case.

Reply:
Well, XML is handled differently; no NUL bytes are allowed: https://3v4l.org/IlUkF

Reply:
Thanks for checking, that makes sense.